designs/provisioner-priority.md

```diff
@@ -0,0 +1,268 @@
+
+## Use Cases
+
+1. Creating a strongly preferred default provisioner that will always be attempted first, except when more specific configuration is required. This enables scenarios like specifying `kubernetes.io/arch=arm64` for workloads that require this architecture, while preferring `kubernetes.io/arch=amd64` on all other workloads, even if the pod does not contain specific `nodeSelector` constraints. This is particularly useful for those that have high numbers of workloads that only work on `amd64` instances, but they do not want to have to go and assign the specific architecture-specific `nodeSelector` to each of these workloads.
+
+   **Example**
+
+    ```yaml
+    # Default provisioner that will be attempted to be scheduled first
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: default
+    spec:
+      weight: 100
+      requirements:
+      - key: kubernetes.io/arch
+        operator: In
+        values: ["amd64"]
+    ```
+
+    ```yaml
+    # ARM-64 specific provisioner that will be scheduled if the other provisioner does not fit the constraints for this provisioner
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: arm64
+    spec:
+      weight: 50
+      requirements:
+      - key: kubernetes.io/arch
+        operator: In
+        values: ["arm64"]
+    ```
+
+2. Creating a strongly preferred default provisioner for enterprise users that have `reserved` instance types and want to prefer having these instance types scheduled prior to other instance types.
+
+   **Example**
+
+    ```yaml
+    # I have c5.large instance types as reserved instances, so I want to schedule to these nodes first
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: reserved
+    spec:
+      weight: 50
+      requirements:
+      - key: "node.kubernetes.io/instance-type"
+        operator: In
+        values: ["c5.large"]
+    ```
+
+    ```yaml
+    # For all other pods, we can use these other instance types
+
+    apiVersion: karpenter.sh/v1alpha5
+    kind: Provisioner
+    metadata:
+      name: fallback
+    spec:
+      requirements:
+      - key: "node.kubernetes.io/instance-type"
+        operator: In
+        values: ["m5.large", "m5.2xlarge"]
+    ```
+
+3. Allowing provisioner with taints to be attempted first so that pods that have tolerations for these taints can be scheduled to specific instance types. Without this ordering, it is possible these pods will be scheduled to nodes that have no tolerations.
```

Comment:
```suggestion
3. Allowing provisioner with taints to be attempted first so that pods that have tolerations for these taints can be scheduled to specific instance types. Without this ordering, it is possible these pods will be scheduled to nodes that have no taints.
```